javascript/src/components/streams/StreamControls.jsx

```diff
@@ -0,0 +1,80 @@
+/* global jsRoutes */
+
+'use strict';
+
+var React = require('react/addons');
+var DropdownButton = require('react-bootstrap').DropdownButton;
+var MenuItem = require('react-bootstrap').MenuItem;
+var ButtonGroup = require('react-bootstrap').ButtonGroup;
+var StreamForm = require('./StreamForm');
+var PermissionsMixin = require('../../util/PermissionsMixin');
+
+var StreamControls = React.createClass({
+    mixins: [PermissionsMixin],
+    getInitialState() {
+        return {};
+    },
+    _onResume(evt) {
+        this.props.onResume(this.props.stream);
+        this.refs.dropdownButton.setDropdownState(false);
+    },
+    _onPause(evt) {
+        this.props.onPause(this.props.stream);
+        this.refs.dropdownButton.setDropdownState(false);
+    },
+    _onEdit(evt) {
+        this.refs.streamForm.open();
+    },
+    _onClone(evt) {
+        this.refs.cloneForm.open();
+    },
+    _onCloneSubmit(streamId, stream) {
+        this.props.onClone(this.props.stream.id, stream);
+    },
+    _onQuickAdd() {
+        this.props.onQuickAdd(this.props.stream.id);
+    },
+    render() {
+        var permissions = this.props.permissions;
+        var stream = this.props.stream;
+
+        var menuItems = [];
+
+        if (this.isPermitted(permissions, ['streams:edit:' + stream.id])) {
+            menuItems.push(<MenuItem key={"editStreams-" + stream.id} onClick={this._onEdit}>Edit stream</MenuItem>);
+            menuItems.push(<MenuItem key={"quickAddRule-" + stream.id}className={stream.stream_rules.length > 0 ? "" : "disabled"}
```

Comment:
Why is the quick add rule option disabled when there are no rules? Maybe it was like that for ages, but it doesn't make much sense to me.